dask/utils.py

```diff
@@ -2091,6 +2092,15 @@ def get_default_shuffle_algorithm() -> str:
         from distributed import default_client
 
         default_client()
+        # We might loose annotations if low level fusion is active
+        if not dask.config.get("optimization.fuse.active"):
```

Comment:
I added this guard to only toggle to p2p if low level fusion is disabled. We are relying on annotations and low level fusion apparently strips them sometimes.